NDTensors/test/test_dense.jl

```diff
@@ -234,18 +234,16 @@ using .NDTensorsTestUtils: devices_list
   # Only CPU backend testing
   @testset "Contract with exotic types" begin
     # BigFloat is not supported on GPU
-    M = Tensor(BigFloat, (1, 2, 3))
-    fill!(M, BigFloat(randn(Float64)))
-    O = Tensor(M, (2, 3))
-    N = Tensor([BigFloat(1.0)], (1,))
-  
-    @test O ≈ contract(M, (-1, 2, 3), N, (-1,))
-    O = similar(M)
-    ## TODO find a way to test the next line, its not possible to access an undefined
-    ## reference
-    NDTensors.contract!!(O, (2, 3), M, (-1, 2, 3), N, (-1,), 0.0, 1.0)
-    NDTensors.contract!!(O, (2, 3), M, (-1, 2, 3), N, (-1,), 0.0, 0.0)
-    @test iszero(O)
+    ## randn(BigFloat, ...) is not defined in Julia 1.6
+    a = BigFloat.(randn(Float64, 2, 3))
+    t = Tensor(a, (1, 2, 3))
+    m = Tensor(a, (2, 3))
+    v = Tensor([one(BigFloat)], (1,))
+
+    @test m ≈ contract(t, (-1, 2, 3), v, (-1,))
+    tp = similar(t)
+    NDTensors.contract!(tp, (1, 2, 3), t, (-1, 2, 3), v, (-1,), 0.0, 0.0)
```

Comment:
Let's use `false` or `zero(Bool)` instead of `0.0` to be more agnostic about the element type (`0.0` is implicitly `Float64` which is pretty opinionated).

Reply:
We might want to also test a case where `iszero(α)` and `!iszero(β)`, in which case you can construct `tp` with random data, for example with `tp = similar(t); randn!(tp)`.

Reply:
Makes sense, I replaced alpha and beta with `true` and `false` instead of `0.0` and `1.0` and I added the case `NDTensors.contract!(tp, (1, 2, 3), t, (-1, 2, 3), v, (-1,), false, BigFloat(randn(Float64)))` to check the scaling option. the `LinearAlgebra` function `scal` is not actually defined on `BigFloat` so I converted to using Broadcast.